File: benchmarks/console/ConsoleWrites.c
#ifdef _WIN32
#define _CRT_SECURE_NO_WARNINGS
#endif

#include <stdio.h>

int main(void) {
    static const int iterations = 64;
    for (int iteration = 0; iteration < iterations; iteration += 1) {
        if (fputs("small", stdout) < 0
            || fputs(" line\n", stdout) < 0
            || fputs("wide \xCE\xB1\n", stdout) < 0
            || fputs("buffer payload 0123456789ABCDEF\n", stdout) < 0
            || fputs("err\n", stderr) < 0) {
            return 1;
        }
    }

    if (fflush(stdout) != 0 || fflush(stderr) != 0) {
        return 2;
    }

    return 0;
}

File: benchmarks/console/ConsoleWrites.rs
use std::io::{self, Write};

fn main() {
    const ITERATIONS: usize = 64;
    let mut stdout = io::stdout().lock();
    let mut stderr = io::stderr().lock();

    for _ in 0..ITERATIONS {
        stdout.write_all(b"small").unwrap();
        stdout.write_all(b" line\n").unwrap();
        stdout.write_all("wide α\n".as_bytes()).unwrap();
        stdout
            .write_all(b"buffer payload 0123456789ABCDEF\n")
            .unwrap();
        stderr.write_all(b"err\n").unwrap();
    }

    stdout.flush().unwrap();
    stderr.flush().unwrap();
}
